include/device.h

```diff
@@ -102,19 +95,18 @@ extern "C" {
  * during initialization.
  */
 #ifndef CONFIG_DEVICE_POWER_MANAGEMENT
-#define DEVICE_AND_API_INIT(dev_name, drv_name, init_fn, data, cfg_info,  \
-			    level, prio, api)				  \
-	static const struct device_config _CONCAT(__config_, dev_name) __used \
-	__attribute__((__section__(".devconfig.init"))) = {		  \
-		.name = drv_name, .init = (init_fn),			  \
-		.config_info = (cfg_info)				  \
-	};								  \
-	static Z_DECL_ALIGN(struct device) _CONCAT(__device_, dev_name) __used \
-	__attribute__((__section__(".init_" #level STRINGIFY(prio)))) = { \
-		.config = &_CONCAT(__config_, dev_name),		  \
-		.driver_api = api,					  \
-		.driver_data = data					  \
-	}
+#define DEVICE_AND_API_INIT(dev_name, drv_name, init_fn, data, cfg_info, \
```

Comment:
i don't fully get what is the difference between `DEVICE_AND_API_INIT` and `DEVICE_DEFINE`. Is it only pm support in `DEVICE_DEFINE`? If so then i naming is confusing. I would prefer to see one macro `DEVICE_DEFINE` if device has no power management then NULL is provided for `pm_control_fn`

Reply:
Yes, these macros would require some rework as well. 
Basically, there should be only one macro: DEVICE_DEFINE(), so it would follow the same naming as the kernel. Let's see this in further PR.

Reply:
sorry, i didn't not notice that you didn't change those macros. it is messy indeed.